File: Hacker Rank/Day of the Programmer.java

import java.time.YearMonth;
import java.util.Scanner;
public class hacker_rank_1 {
    public static  void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int val = sc.nextInt();
        int limit = 256;

        if(val < 1919){
            int sum = 0;
            for(int i = 1; i <= 12; i++){
                sum = sum +  numberOfDaysInMonth(val, i);

                if(limit - sum < 30){
                    int date = limit-sum;
                    System.out.println(date+"."+String.format("%02d", i+1)+"."+val);
                    break;
                }

            }
        }
        else
        {
                int sum = 0;
                for(int i = 1; i <= 12; i++){
                    sum = sum +  numberOfDaysInMonth(val, i);

                    if(limit - sum < 30){
                        int date = limit-sum;
                        System.out.println(date+"."+String.format("%02d", i+1)+"."+val);
                        break;
                    }

                }
        }
    }
    public static int numberOfDaysInMonth(int year, int month){
        YearMonth yearMonth  =YearMonth.of(year, month);
        int daysInMonth = yearMonth.lengthOfMonth();
        return  daysInMonth;
    }
}




// https://www.hackerrank.com/challenges/java-stack/problem?isFullScreen=true

File: Hacker Rank/Kotlin/Day of the Programmer.kt
import java.time.YearMonth
import java.util.*

fun main(args: Array<String>){
    var sc = Scanner(System.`in`)

    val year = sc.nextInt()
    var limit = 256

    if(year<1919) {
        var sum = 0
        for (i in 1..12) {
            sum = sum + numberOfDaysInMonth(year, i)
            if (limit - sum < 30) {
                var date = limit - sum
                println(date.toString() + "." + String.format("%02d", i + 1) + "." +year)
                break
            }
        }
    }
    else{
        var sum = 0
        for (i in 1..12) {
            sum = sum + numberOfDaysInMonth(year, i)
            if (limit - sum < 30) {
                var date = limit - sum
                println(date.toString() + "." + String.format("%02d", i+1) + "." +year)
                break
            }
        }
    }


}

fun numberOfDaysInMonth(year: Int , month:Int):Int
{
    var yearMonth: YearMonth = YearMonth.of(year, month)
    var daysInMonth = yearMonth.lengthOfMonth()
    return daysInMonth
}
